src/backend/routers/agent.py

```diff
@@ -336,13 +352,23 @@ async def list_agent_tool_metadata(
     Raises:
         HTTPException: If the agent tool metadata retrieval fails.
     """
+    user_id = ctx.get_user_id()
+    tool_metadata = []
+    agent = None
+
     try:
-        return agent_tool_metadata_crud.get_all_agent_tool_metadata_by_agent_id(
-            session, agent_id
+        agent = agent_crud.get_agent_by_id(session, agent_id)
+        tool_metadata = (
+            agent_tool_metadata_crud.get_all_agent_tool_metadata_by_agent_id(
+                session, agent_id
+            )
         )
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e))
 
+    validate_user_has_access_to_agent(user_id, agent, agent_id=agent_id)
```

Comment:
Could we not do this in `get agent by id`?

Reply:
I was trying to keep the crud functions atomic, but yeah I can change the filter there